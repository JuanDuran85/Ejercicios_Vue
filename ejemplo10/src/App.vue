<template>
  <div id="app">
    <div class="titulo">
      <h1>Random Git Cat</h1>
    </div>
    <div class="formulario">
      <label for="">
        Titulo: <input type="text" v-model="titulo">
      </label>
      <label for="filtro">Filtro
        <select name="filtro" id="filtro" v-model="filtro">
          <option value="blur">blur</option>
          <option value="mono">mono</option>
          <option value="sepia">sepia</option>
          <option value="negative">negative</option>
          <option value="paint">paint</option>
          <option value="pixel">pixel</option>
        </select>
      </label>
      <label for="color">Color
        <select name="color" id="color" v-model="color">
          <option value="red">Rojo</option>
          <option value="blue">Azul</option>
          <option value="green">Verde</option>
          <option value="yellow">Amarillo</option>
          <option value="white">Blanco</option>
        </select>
      </label>
      <label for="">
          Tama√±o<input type="number" v-model="medida">
      </label>
    </div>
    <div class="muestra">
      <button @click.prevent="buscar">Mostrar Gato</button>
      <componente-imagen :src="buscarURL"/>
    </div>
  </div>
</template>

<script>
import CompoImg from './components/CompoImg.vue'

export default {
  name: 'App',
  data() {
    return {
      titulo: "Escribe el nombre",
      color: "blue",
      medida: "300",
      filtro: "blur",
      resultado: [],
    }
  },
  methods: {
    buscar(){
      console.log("entro");
      fetch(`https://cataas.com/cat/gif/says/${this.titulo}?filter=${this.filtro}&color=${this.color}&size=${this.medida}`)
      .then(respuesta => {
          this.resultado = respuesta;
        })
      .catch(error => console.error(error));
    }
  },
  components: {
    "componente-imagen": CompoImg 
  },
  computed: {
    buscarURL(){
      return this.resultado.url;
    }
  },
}
</script>

<style scope>
  .titulo {
    margin-top: -25px;
    font-size: 24px;
    font-weight: 800;
    text-align: center;
    background: rgba(136, 162, 216, 0.795);
    height: 20vh;
  }
  .titulo h1 {
    padding-top: 35px;
  }
  .formulario {
    display: block;
    padding: 5px;
    background: rgba(218, 79, 79, 0.795);
    font-size: 18px;
    text-align: center;
  }
  .formulario label {
    display: block;
    margin: 10px;
    padding: 10px;
  }
  .muestra {
    text-align: center;
    padding: 10px;
    background: rgba(136, 162, 216, 0.795);
    height: auto;
  }
</style>
