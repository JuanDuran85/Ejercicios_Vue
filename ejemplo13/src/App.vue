<template>
  <div id="app">
    <h1>Utilizando Vue-Router con Vue-Cli</h1>
    <nav>
      <ul>
        <li><router-link to="/">Inicio</router-link></li>
        <li><router-link to="/servicios">Servicios</router-link></li>
        <li><router-link to="/blog/1">Blog</router-link></li>
        <li><router-link to="/demo">Demo</router-link></li>
        <li><router-link to="/demo2">Demo2</router-link></li>
        <li><router-link :to="{ name: 'contacto' }">Contacto</router-link></li>
      </ul>
    </nav>
    <router-view></router-view>
  </div>
</template>

<script>

export default {
  name: 'App',
}
</script>

<style>
  body {
    text-align: center;
    font-family: 'Times New Roman', Times, serif;
    font-size: 20px;
    font-weight: 600;
    text-decoration: none;
    list-style: none;
  }
</style>
