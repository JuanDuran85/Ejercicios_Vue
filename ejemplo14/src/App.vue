<template>
  <div>
    <nav class="navbar navbar-light navbar-expand-lg fixed-top" id="mainNav">
        <div class="container"><a class="navbar-brand">V8</a><button data-toggle="collapse" data-target="#navbarResponsive" class="navbar-toggler" aria-controls="navbarResponsive" aria-expanded="false" aria-label="Toggle navigation"><i class="fa fa-bars"></i></button>
            <div
                class="collapse navbar-collapse" id="navbarResponsive">
                <ul class="nav navbar-nav ml-auto" id="menu-v8">
                    <li class="nav-item" role="presentation">
                      <router-link :to="{name: 'Portada'}">Portada</router-link> 
                    </li>
                    <li class="nav-item" role="presentation">
                      <router-link :to="{name: 'SobreMi'}">Sobre Mi</router-link> 
                    </li>
                    <li class="nav-item" role="presentation">
                      <router-link :to="{name: 'Contacto'}">Contacto</router-link>  
                    </li>
                    <li class="nav-item" role="presentation">
                      <router-link to="/post/1">Ãšltimo Post</router-link> 
                    </li>
                </ul>
            </div>
        </div>
    </nav>
    <transition name="efectos">
      <router-view></router-view>
    </transition>
  </div>
</template>

<script>
export default {
  name: 'App'
}
</script>

<style scoped>
#app {
  font-family: 'Avenir', Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;  
}
#menu-v8 a{
  text-shadow: 1px 1px #000;
}
.efectos-enter-active,  
.efectos-leave-active {
  transition: opacity .1s;
}
.efectos-enter, 
.efectos-leave-to {
  opacity: 0;

}
</style>
