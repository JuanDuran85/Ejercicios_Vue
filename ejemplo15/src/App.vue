<template>
  <div id="app">
      <img src="https://lyl-ingenieria.com/blog/wp-content/uploads/2017/08/INTEGRA-WAREHOUSE.jpg" alt="almacen">
      <h1 v-text="titulo"></h1>
      <hr>
      <p v-text="ocupado"></p>
      <p v-text="totalVentas"></p>
      <p v-text="totalDeudas"></p>
      <p v-text="creditos"></p>
      <h3>Productos Disponibles en el almacen:</h3>
      <ul>
        <li v-for="(item, index) in productosConCantidad" :key="index">
          <label v-text="item.name"></label>
        </li>
      </ul>
      <h3>Productos Baratos Disponibles en el almacen:</h3>
      <ul>
        <li v-for="(item, index) in productosConCantidadBaratos" :key="index">
          <label v-text="item.name"></label>
        </li>
      </ul>
      <h3>Busqueda de producto por Identificador: </h3>
      <input type="text" v-model="buscarID">
      <ul v-if="productoPorId.length > 0">
        <li v-for="(item, index) in productoPorId" :key="index">
          <label v-text="item.name"></label>
        </li>
      </ul>
      <p v-else>...</p>
      <hr>
      <p>
        <label v-text="ocupado"></label>
        <input type="button" value="Cambiar el estado" @click="modificandoOcupado">
      </p>
      <hr>
      <hr>
      <p>
        <label v-text="creditos"></label>
        <input type="button" value="Cambiar el estado" @click="modificandoCredito">
      </p>
      <hr>
      <h3>Productos</h3>
      <ul>
        <li v-for="(item, index) in todosLosProductos" :key="index">
          <span>{{item.id}} |  {{item.name}} |  {{item.stock}} | {{item.precio}} |</span>
          <button @click="agregandoStock(item.id)">+</button> 
          <button @click="actualizarVentaProducto(item)">Vender</button>
        </li>
      </ul>
  </div>
</template>

<script>
// se llaman de esta forma porque son objetos de vuex
import {mapState, mapGetters, mapMutations, mapActions} from 'vuex';

export default {
  name: 'App',
  data() {
    return {
      buscarID: ""
    }
  },
  methods: {
    ...mapMutations(['modificandoOcupado']),
    ...mapActions(['modificandoCredito']),
    agregandoStock(productoEnviado){
      this.$store.commit('agregandoStock',productoEnviado);
    },
    actualizarVentaProducto(productoVendido){
      this.$store.dispatch('actualizarVentaProducto',productoVendido)
    }
  },
  computed: {
    titulo(){
      return "Almecen Web";
    },
    ...mapGetters([
      'productosConCantidad'
    ]),
    productosConCantidadBaratos(){
      return this.$store.getters.productosConCantidadBaratos;
    },
    productoPorId(){
      return this.$store.getters.productoPorId(this.buscarID);
    },
    ...mapState({
      todosLosProductos(state){
        return state.productos;
      },
      ocupado(){
        // Al proporcionar la opción de tienda a la instancia raíz, la tienda se inyectará en todos los componentes secundarios de la raíz y estará disponible en ellos como this.$store
        let ocupado = this.$store.state.ocupado ? "Ocupado":"Disponible";
        return `Estado: ${ocupado}`;
      },
      creditos(state){
        return state.creditos ? "Tiene Credito":"No tiene credido";
      },
      totalVentas(){
        console.log(this.$store.state.totalVentas);
        return `Total Ventas: $${this.$store.state.totalVentas}`;
      },
      totalDeudas(state){
        return `Total Deudas: $${state.totalDeudas}`;
      }
    }),
  }
};
</script>

<style>
#app {
  font-size: 20px;
  background-color: azure;
}

#app img {
  width: 300px;
}
</style>
