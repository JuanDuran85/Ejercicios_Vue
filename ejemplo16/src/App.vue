<template>
  <div id="app">
    <transition name="vista">
      <router-view :to="{name: 'Inicio'}"></router-view>
    </transition>
  </div>
</template>

<script>
export default {
  name: 'App',
}
</script>

<style scoped>
.vista-enter-active, .vista-leave-active {
    transition: opacity .1s;
}
.vista-enter, .vista-leave-to{
    opacity: 0;
}
</style>
