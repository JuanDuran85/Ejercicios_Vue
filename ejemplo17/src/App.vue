<template>
  <div id="app">
    <nav class="navbar">
      <a><router-link :to="{name: 'Inicio'}">Incio</router-link></a>
      <a><router-link :to="{name: 'Busqueda'}">BÃºsqueda</router-link></a>
      <a><router-link :to="{name: 'Ventas'}">Ventas</router-link></a>
      <a><router-link :to="{name: 'Busqueda'}">Total</router-link></a>
    </nav>
    <transition name="vista">
      <router-view></router-view>
    </transition>
  </div>
</template>

<script>
export default {
  name: 'App',
}
</script>

<style scoped>
.navbar {
  display: flex;
  background-color: #333;
}
.navbar a {
  color: white;
  padding: 14px 20px;
  text-decoration: none;
  text-align: center;
}
.navbar a:hover {
  background-color: #ddd;
  color: black;
}
.vista-enter-active, .vista-leave-active {
    transition: opacity .1s;
}
.vista-enter, .vista-leave-to{
    opacity: 0;
}
</style>
