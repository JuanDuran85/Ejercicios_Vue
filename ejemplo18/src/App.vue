<template>
  <div>
    <h1>Usando propiedades computadas</h1>
    <p>Ejercicios basicos de funcionamiento</p>
    <hr>
    <p>Ejemplo 1</p>
    Nombre: <input type="text" v-model="nombre"><br>
    Apellido: <input type="text" id="apellido" value="Duran"><br>
    <button @click="guardar">Guardar Apellido</button>
    <output>{{cambiarDatos}}</output>
    <hr>
    <p>Ejemplo 2</p>
    <table class="table table-striped">
    <thead>
        <tr>
            <th>#</th>
            <th>Capital</th>
            <th>Pais</th>
            <th>Poblaci√≥n</th>
        </tr>
    </thead>
    <tbody>
        <tr v-for="(item, index) in poblacionOrdenada" :key="index">
            <th>{{index+1}}</th>
            <td>{{item.capital}}</td>
            <td>{{item.pais}}</td>
            <td>{{item.poblacion}}</td>
        </tr>
    </tbody>
    </table>
  </div>
</template>

<script>
let apellido = "Duran";

export default {
    name: 'App',
    data() {
        return {
            nombre: "Juan",
            datos: [
                {capital:"Caracas",pais:"Venezuela",poblacion:"30000000"},
                {capital:"Santiago de Chile",pais:"Chile",poblacion:"19000000"},
                {capital:"Bogota",pais:"Colombia",poblacion:"50000000"},
                {capital:"Peru",pais:"Lima",poblacion:"20000000"},
            ]
        }
    },
    methods: {
        guardar(){
            apellido = this.$el.querySelector("#apellido").value;
        }
    },
    computed: {
        cambiarDatos(){
            return `${this.nombre} ${apellido}`;
        },
        poblacionOrdenada(){
            return this.datos.sort((v1,v2) => v2.poblacion - v1.poblacion);
        }
    },
}
</script>

<style>
div {
    background-color: azure;
    font-size: 18px;
}
</style>