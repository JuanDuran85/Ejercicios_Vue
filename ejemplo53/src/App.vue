<template>
  <div id="app">
    <navegation  :users="users" @logout="logout"></navegation>
    <router-view class="container" :meetings="meetings" :users="users" @logout="logout" @addMeeting="addMeeting" @deleteMeeting="deleteMeeting" @checkIn="checkIn" :error="error"/>
    <div v-if="activo" class="d-flex justify-content-center mt-5 pt-3">
      <div class="spinner-border text-info" style="width: 3rem; height: 3rem;" role="status">
        <span class="sr-only">Loading...</span>
      </div>
    </div>
  </div>
</template>

<script>
import Navegation from '@/components/Navegation.vue';
import firebase from 'firebase';
import {db} from './main';

export default {
  name: 'App',
  components: {
    Navegation
  },
  data() {
    return {
      users: [],
      meetings: [],
      error: [],
      activo: false
    }
  },
  methods: {
    logout(){
      firebase.auth().signOut().then(()=>{
        this.users = [];
        this.meetings = [];
        this.$router.push('Login');
      })
    },
    addMeeting(recibido){
      db.collection("users").doc(this.users.uid).collection("meetings").add({
        name: recibido,
        createdAt: firebase.firestore.FieldValue.serverTimestamp()
      });
      alert("Cargado en base de datos");
    },
    deleteMeeting(id){
      db.collection("users").doc(this.users.uid).collection("meetings").doc(id).delete();
    },
    checkIn(datos){
      this.activo = true;
      console.log(datos);
      db.collection("users").doc(datos.userID).collection("meetings").doc(datos.meetingsID).get().then(documento=>{
        if(documento.exists){
          db.collection("users").doc(datos.userID).collection("meetings").doc(datos.meetingsID).collection("atenciones").add({
            name: datos.name,
            email: datos.email,
            createdAt: firebase.firestore.FieldValue.serverTimestamp()
          }).then(()=> {
            this.activo = false;
            this.$router.push('/attendees/'+datos.userID+'/'+datos.meetingsID);
          })
        }else{
          this.error.push('Lo sentimos, no existe el evento')
        }
      })
    }
  },
  mounted() {
    firebase.auth().onAuthStateChanged(user => {
      if (user){
        //this.users = user.email;
        //this.users = user.displayName;
        this.users = user; //pasando toda la informacion del usuario
        db.collection("users").doc(this.users.uid).collection("meetings").onSnapshot(result=>{
          let auxiliar = [];
          result.forEach(doc=>{
            auxiliar.push({
              id: doc.id,
              name: doc.data().name
            })
          });
          this.meetings = auxiliar.sort((a,b)=>{
            if (a.name.toLowerCase() < b.name.toLowerCase())
            return -1
            return 1
          })
        })
      }
    });
/*     db.collection("users").doc("P2pXb9II3lPAEVoGGmSu").get().then(response => {
      console.log(response.data())
      this.users = response.data().name;
    })  */   
  },  
}
</script>

<style lang="scss">
$primary: #05b2dd;
@import "../node_modules/bootstrap/scss/bootstrap";
</style>
