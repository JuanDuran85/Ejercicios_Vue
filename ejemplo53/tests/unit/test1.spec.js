import { shallowMount } from '@vue/test-utils';
import Vue from 'vue'
import router from '@/router'
import App from '@/App';

describe('App.vue', () => {
    test('Todo bien', () => {
        const wrapper = shallowMount(App)
        expect(wrapper.get('#nav'))
        expect(wrapper.find('div'))
        expect(wrapper.element).toMatchSnapshot()
        expect(wrapper.text()).toMatch(`Home`)
    })
})