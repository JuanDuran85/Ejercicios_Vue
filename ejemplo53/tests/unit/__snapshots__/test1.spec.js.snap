// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue Todo bien 1`] = `
<div
  id="app"
>
  <div
    id="nav"
  >
    <router-link-stub
      ariacurrentvalue="page"
      event="click"
      tag="a"
      to="/"
    >
      Home
    </router-link-stub>
     |
    
    <router-link-stub
      ariacurrentvalue="page"
      event="click"
      tag="a"
      to="/about"
    >
      About
    </router-link-stub>
  </div>
   
  <router-view-stub
    name="default"
  />
</div>
`;
